nltk/sem/relextract.py

```diff
@@ -198,17 +198,20 @@ def extract_rels(subjclass, objclass, doc, corpus='ace', pattern=None, window=10
     :return: see ``mk_reldicts``
     :rtype: list(defaultdict)
     """
-
-    if subjclass and subjclass not in NE_CLASSES[corpus]:
-        if _expand(subjclass) in NE_CLASSES[corpus]:
-            subjclass = _expand(subjclass)
-        else:
-            raise ValueError("your value for the subject type has not been recognized: %s" % subjclass)
-    if objclass and objclass not in NE_CLASSES[corpus]:
-        if _expand(objclass) in NE_CLASSES[corpus]:
-            objclass = _expand(objclass)
-        else:
-            raise ValueError("your value for the object type has not been recognized: %s" % objclass)
+    
+    for i, s in enumerate(subjclass):
+        if s and s not in NE_CLASSES[corpus]:
+            if _expand(s) in NE_CLASSES[corpus]:
+                subjclass[i] = _expand(s)
+            else:
+                raise ValueError("your value for the subject type has not been recognized: %s" % s)
+
+    for i, o in enumerate(objclass):
```

Comment:
Lines 202-207 and 210-214 seem to do the same thing but on different variables. If the code is moved to a function, that that function has to be called with different argument.

I'm talking about the `ValueError` (lines 214 and 207), the code i've written doesn't do such a check.